### PR TITLE
protections public methods

### PR DESCRIPTION
**Pull request by baptiste-grob**
With this PR we'll end up with:

- `SNApplication.hasProtections()`: to determine if a user can use the feature at all (which they can't if they don't have an account, passcode, or biometrics authentication)
- `SNApplication.areProtectionsEnabled()`: to determine if protections are enabled/disabled

### CODE REVIEW

**Comment by mobitar** [comment]
Left some comments here: https://github.com/standardnotes/snjs/pull/224/files#r577674680

**Comment by baptiste-grob** [comment]
Should be good to review now @mobitar @antsgar :)

File: packages/snjs/lib/application.ts [hunk 1]
```diff
@@ -888,6 +888,22 @@ export class SNApplication {
     return this.protocolService.hasAccount();
   }
 
+  /**
+   * Use this method if you do not want to display protections-related
+   * information to a user who cannot currently exploit the feature.
+   */
+  public hasProtectionSources(): boolean {
```
Comment (mobitar) [inline packages/snjs/lib/application.ts:895]: What do you think of a comment like this:

```
Returns true if the user has a source of protection available, such as a passcode, password, or biometrics.
```

I find the current comment somewhat awkwardly phrased. "Use..if you do not..who cannot..exploit"..

**Review by antsgar — APPROVED** [review]

File: packages/snjs/lib/application.ts (hunk 1 above)
Comment (baptiste-grob) [reply packages/snjs/lib/application.ts:895]: Sounds good! I was struggling with this one

**Pull request merged**